### ISSUE TITLE
Display of changed data

### ISSUE THREAD

**Issue by BenBenna** [gino0631/nextcloud-metadata#36]
I put this question already forward in the NC-Forum - but got no answer.

Whenever the metadata content of a .jpg file (e.g. Description, Keyword, etc) is changed in win10 explorer this metadata get no longer displayed in nextcloud.

I get this error:

    [PHP] Error: array_key_exists() expects parameter 2 to be array, boolean given at /var/www/nextcloud/apps/metadata/lib/Controller/IptcMetadata.php#29

    GET /apps/metadata/get?source=%2FPhotos%2FiPhone6%2F2019%2FIMG_7277.JPG
    from 91.46.190.236 by DerberMohikaner at 2019-07-31T20:01:30+02:00

The "wheel" rotates and rotates and there is no display of metadate on this file never again.

Nextcloud is Version 16.0.3 and is operated on an Odroid N2.

Is there a way to avoid this?
Thank you

**Comment by gino0631** [comment]
Could you please provide a sample of such .jpg to reproduce the problem?

**Comment by BenBenna** [comment]
Thanks for your immediate response.
The error is:
{"reqId":"VL0Uo2pgtPgFX3yOZClg","level":3,"time":"2019-08-03T13:34:51+02:00","remoteAddr":"84.166.76.125","user":"DerberMohikaner","app":"index","method":"GET","url":"/apps/metadata/get?source=%2FPhotos%2FiPhone6%2F2019%2FIMG_7278.JPG","message":{"Exception":"Exception","Message":null,"Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":119,"function":"render","class":"OCP\\AppFramework\\Http\\JSONResponse","type":"->","args":[]},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":126,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Metadata\\Controller\\MetadataController"},"get"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Metadata\\Controller\\MetadataController","get",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"metadata.metadata.get"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"metadata.metadata.get"}]},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"metadata.metadata.get"}]},{"file":"/var/www/nextcloud/lib/base.php","line":975,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/metadata/get"]},{"file":"/var/www/nextcloud/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/nextcloud/lib/public/AppFramework/Http/JSONResponse.php","Line":71,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0","version":"16.0.3.0","id":"5d4571b597215"}

{"reqId":"VL0Uo2pgtPgFX3yOZClg","level":3,"time":"2019-08-03T13:34:51+02:00","remoteAddr":"84.166.76.125","user":"DerberMohikaner","app":"PHP","method":"GET","url":"/apps/metadata/get?source=%2FPhotos%2FiPhone6%2F2019%2FIMG_7278.JPG","message":"array_key_exists() expects parameter 2 to be array, boolean given at /var/www/nextcloud/apps/metadata/lib/Controller/IptcMetadata.php#29","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0","version":"16.0.3.0","id":"5d4571b5972a8"}
![IMG_7264_Metadata](https://user-images.githubusercontent.com/14091725/62411875-496f6780-b5fa-11e9-93f9-93fcf82ae3e9.PNG)
![IMG_7264_Metadata](https://user-images.githubusercontent.com/14091725/62411879-6ad05380-b5fa-11e9-83cf-f4f1dfb9930b.PNG)
![7264_Activity](https://user-images.githubusercontent.com/14091725/62411880-6b68ea00-b5fa-11e9-9443-3c5733f7a023.PNG)
![NC-Activity_7264](https://user-images.githubusercontent.com/14091725/62411881-6b68ea00-b5fa-11e9-9751-9de4fb1ef1b4.PNG)
![win10_7268-2](https://user-images.githubusercontent.com/14091725/62411882-6b68ea00-b5fa-11e9-80b0-552f079c6ad7.PNG)
![win10_7268](https://user-images.githubusercontent.com/14091725/62411883-6b68ea00-b5fa-11e9-8149-c15e99b713d5.PNG)
![NC_7268](https://user-images.githubusercontent.com/14091725/62411884-6c018080-b5fa-11e9-9e73-b01d921deb14.PNG))

**Comment by BenBenna** [comment]
Unfortunately i do not understand how to answer correvctly.

My problem is that after i  change a detail on a given image, the data of this image is updated (created / updated), but the changed metadata (comment / description, etc) is not displayd anymore. 
I only see the rotating "wheel of eternity"

**Comment by gino0631** [comment]
Could you please attach the modified .jpg file?

**Comment by BenBenna** [comment]
This is a complete print out.
[index] Error: Exception: null at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 119
   OCP\AppFramework\Http\JSONResponse->render()
1. /var/www/nextcloud/lib/private/AppFramework/App.php line 126
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Metadata\Con ... {}, "get")
2. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Metadata\\ ... r", "get", OC\AppFramework\ ... {}, {_route: "metadata.metadata.get"})
3. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "metadata.metadata.get"})
4. /var/www/nextcloud/lib/private/Route/Router.php line 297
   undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {_route: "metadata.metadata.get"})
5. /var/www/nextcloud/lib/base.php line 975
   OC\Route\Router->match("/apps/metadata/get")
6. /var/www/nextcloud/index.php line 42
   OC::handleRequest()

GET /apps/metadata/get?source=%2FPhotos%2FiPhone6%2F2019%2FIMG_7264.JPG
from 84.166.76.125 by DerberMohikaner at 2019-08-03T20:12:00+02:00
![IMG_7264](https://user-images.githubusercontent.com/14091725/62415540-94ee3980-b62b-11e9-8b7a-12d537bd82ab.JPG)

Thank you very much for your interest and help

**Comment by gino0631** [comment]
@BenBenna the issue should be fixed in the current master. Would it be possible for you to test (by downloading the current master from GitHub and putting it to the apps folder)?

**Comment by BenBenna** [comment]
Sorry for my late reply.
This is due to me trying to download and test the new version. I did not succeed to get the folder in place with due to my poor knowledge of PuTTY. Sorry. I have tried PuTTY, WinSCP and FileZilly but I did not overcome the trivial problem posed by limited file restrictions. So i will have to wait for the next release or a sudden enlightenment.
Thank you very much. If there is an alternative way to Paypal to support your work i will gladly do.
